src/Desktop.Analyzers/Core/DoNotUseInsecureXSLTScriptExecution.cs

```diff
@@ -0,0 +1,313 @@
+                                                                                                                          HelpLink);
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> supportDiagnostics = ImmutableArray.Create(RuleXslCompiledTransformLoadInsecureInputSettings,
+                                                                                                                RuleXslCompiledTransformLoadInsecureConstructedSettings);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return DoNotUseInsecureXSLTScriptExecutionAnalyzer<TLanguageKindEnum>.supportDiagnostics;
+            }
+        }
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+
+            analysisContext.RegisterCompilationStartAction(
+                (context) =>
+                {
+                    var compilation = context.Compilation;
+                    var xmlTypes = new CompilationSecurityTypes(compilation);
+                    if (xmlTypes.XslCompiledTransform != null)
+                    {
+                        context.RegisterCodeBlockStartAction<TLanguageKindEnum>(
+                            (c) =>
+                            {
+                                GetAnalyzer(c, xmlTypes);
+                            });
+                    }
+                });
+        }
+
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(LocalizableResourceString messageFormat, LocalizableResourceString description, string helpLink = null)
+        {
+            return new DiagnosticDescriptor(RuleId,
+                                            SecurityDiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.InsecureXsltScriptProcessingMessage)),
+                                            messageFormat,
+                                            DiagnosticCategory.Security,
+                                            DiagnosticSeverity.Warning,
+                                            isEnabledByDefault: true,
+                                            description: description,
+                                            helpLinkUri: helpLink,
+                                            customTags: WellKnownDiagnosticTags.Telemetry);
+        } 
+
+        protected abstract Analyzer GetAnalyzer(CodeBlockStartAnalysisContext<TLanguageKindEnum> context, CompilationSecurityTypes types);
+
+        protected class Analyzer
+        {
+            private readonly CompilationSecurityTypes xmlTypes;
+            private readonly SyntaxNodeHelper syntaxNodeHelper;
+
+            private readonly Dictionary<ISymbol, XsltSettingsEnvironment> xsltSettingsEnvironments = new Dictionary<ISymbol, XsltSettingsEnvironment>();
```

Comment:
Please follow the corefx coding style
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md